### PR TITLE
Create access token to upload batch specs from src-cli in executor

### PR DESCRIPTION
**Pull request by mrnugget**
This PR does two things:

* Set the env vars required to run the executor locally. We need a proper fix for this. See https://github.com/sourcegraph/sourcegraph/issues/22550
* It creates an access token for each batch spec execution. This is _not ideal_. See the code comment. We'll add this to the follow-up work for #22550. Until then it doesn't hurt anybody, except our database locally. So to get something working we add this and then we build a proper solution.

### CODE REVIEW

File: enterprise/cmd/executor-queue/internal/queues/batches/transform.go [hunk 1]
```diff
@@ -34,9 +65,7 @@ func transformRecord(ctx context.Context, db dbutil.DB, exec *btypes.BatchSpecEx
 					"-text-only",
 				},
 				Dir: ".",
-				Env: []string{
-					fmt.Sprintf("SRC_ENDPOINT=%s", srcEndpoint),
-				},
+				Env: cliEnv,
 			},
 		},
 		RedactedValues: map[string]string{
```
Comment (eseliger) [inline enterprise/cmd/executor-queue/internal/queues/batches/transform.go:71]: token should be in here

File: enterprise/cmd/executor-queue/internal/queues/batches/transform.go [hunk 2]
```diff
@@ -55,6 +84,19 @@ func transformRecord(ctx context.Context, db dbutil.DB, exec *btypes.BatchSpecEx
...
+	accessTokenScope = "user:all"
+)
+
+func createAccessToken(ctx context.Context, db dbutil.DB, userID int32) (string, error) {
+	_, token, err := database.AccessTokens(db).Create(ctx, userID, []string{accessTokenNote}, accessTokenNote, userID)
+	if err != nil {
+		return "", err
+	}
+	return token, err
+}
```
Comment (eseliger) [inline enterprise/cmd/executor-queue/internal/queues/batches/transform.go:98]: I think we should clean these up at some point. (can be later, no need in this PR, but we should track it)

**Review by eseliger — APPROVED** [review]

File: enterprise/cmd/executor-queue/internal/queues/batches/transform.go (hunk 1 above)
Comment (mrnugget) [reply enterprise/cmd/executor-queue/internal/queues/batches/transform.go:71]: Done!

File: enterprise/cmd/executor-queue/internal/queues/batches/transform.go (hunk 2 above)
Comment (mrnugget) [reply enterprise/cmd/executor-queue/internal/queues/batches/transform.go:98]: See: https://github.com/sourcegraph/sourcegraph/issues/22550

**Comment by mrnugget** [comment]
@sourcegraph/batchers We're going to merge this since it's little code and we want to get to a first deployment on k8s.sgdev.org as possible.

**Pull request merged**